### PR TITLE
Fix snap

### PR DESCRIPTION
**Pull request by sergiusens**
- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

**Review by cjp256 — APPROVED** [review]

**Comment by cjp256** [comment]
It could be <3.5, as I read that cryptography will require rust after that.  Maybe 3.4 will be maintained for a long time?

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -83,12 +83,17 @@ parts:
   snapcraft:
     source: .
     plugin: python
+    python-packages:
+        # Lock down cryptography to 3.4 to override the rust dependency.
+        - cryptography==3.4
```
Comment (cjp256) [inline snap/snapcraft.yaml:88]:
```suggestion
        - cryptography<3.5
```

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -114,11 +119,16 @@ parts:
 
   legacy-snapcraft:
     plugin: python
+    python-packages:
+        # Lock down cryptography to 3.4 to override the rust dependency.
+        - cryptography==3.4
```
Comment (cjp256) [inline snap/snapcraft.yaml:124]:
```suggestion
        - cryptography<3.5
```

**Review by cjp256 — APPROVED** [review]

**Comment by xnox** [comment]
The local setup script was not adjusted to do the same, and ditto the requirements.txt When bootstrapping this on riscv, rustc is neither installed, nor is cryptography built without rust.

We could just build-dep on rust toolchain and upgrade to new cryptography however, as rust is available everywhere.... Any thoughts?

**Comment by nanonyme** [comment]
@cjp256 what makes you think 3.4 will be maintained for long time? Have you gotten explicit promise about that from someone?

**Pull request merged**